plugins/animate/spin.hpp

```diff
@@ -0,0 +1,100 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2024 Scott Moreau <oreaus@gmail.com>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include "animate.hpp"
+#include "wayfire/toplevel-view.hpp"
+#include <memory>
+#include <wayfire/plugin.hpp>
+#include <wayfire/opengl.hpp>
+#include <wayfire/view-transform.hpp>
+#include <wayfire/output.hpp>
+
+static const std::string spin_transformer_name = "spin-transformer";
+namespace wf
+{
+namespace spin
+{
+using namespace wf::animation;
+class spin_animation_t : public duration_t
+{
+  public:
+    using duration_t::duration_t;
+};
+class spin_animation : public animation_base
+{
+    wayfire_view view;
+    wf_animation_type type;
+    wf::spin::spin_animation_t progression;
+
+  public:
+
+    void init(wayfire_view view, wf::animation_description_t dur, wf_animation_type type) override
+    {
+        this->view = view;
+        this->type = type;
+        this->progression =
+            wf::spin::spin_animation_t(wf::create_option<wf::animation_description_t>(dur));
+
+        if (type & HIDING_ANIMATION)
+        {
+            this->progression.reverse();
+        }
+
+        this->progression.start();
+
+        auto tr = std::make_shared<wf::scene::view_2d_transformer_t>(view);
+        view->get_transformed_node()->add_transformer(
+            tr, wf::TRANSFORMER_HIGHLEVEL, spin_transformer_name);
+    }
+
+    bool step() override
+    {
+        auto transform = view->get_transformed_node()
+            ->get_transformer<wf::scene::view_2d_transformer_t>(spin_transformer_name);
+        auto progress = this->progression.progress();
+        transform->alpha   = progress;
+        transform->angle   = progress * M_PI * 4.0;
```

Comment:
Idea: let the user configure the `4.0` constant, that way they can configure how often the window will rotate before closing/opening completely.

Reply:
Already way ahead of you, the [wayfire-animations branch](https://github.com/soreau/wayfire/blob/wayfire-animations/plugins/animate/spin.hpp#L82-L83) has a version which provides the feature you request.

Reply:
great :)